lms/celery.py

```diff
@@ -25,14 +24,35 @@
 APP.autodiscover_tasks(lambda: settings.INSTALLED_APPS)
 
 
-class Router(AlternateEnvironmentRouter):
-    """
-    An implementation of AlternateEnvironmentRouter, for routing tasks to non-cms queues.
+def route_task(name, args, kwargs, options, task=None, **kw):  # pylint: disable=unused-argument
     """
+    Celery-defined method allowing for custom routing logic.
 
-    @property
-    def alternate_env_tasks(self):
-        """
-        Defines alternate environment tasks, as a dict of form { task_name: alternate_queue }
-        """
-        return {}
+    If None is returned from this method, default routing logic is used.
+    """
+    # Defines alternate environment tasks, as a dict of form { task_name: alternate_queue }
+    alternate_env_tasks = {}
+
+    # Defines the task -> alternate worker queue to be used when routing.
+    explicit_queues = {
+        'openedx.core.djangoapps.content.course_overviews.tasks.async_course_overview_update': {
+            'queue': settings.GRADES_DOWNLOAD_ROUTING_KEY},
```

Comment:
Did these need to be added because `routing_key` on the `@task` decorator and `Task` objects is deprecated?  If so, should it also be removed from the decorators?

Reply:
@jmbowman You have to mention the `queue` parameter in the `@task` decorator. We have verified this in this PR https://github.com/edx/edx-platform/pull/25314. We can either go by adding the `queue` parameter in the `@task` decorator or by removing `rotuing_key` from there and keep routing in `explicit_queues` dict. Which approach should I go with?

Reply:
I have a mild preference for keeping the queue assignments here (one step towards configuring this in settings rather than hard-coding), so we just nee to remove the `routing_key` attributes.